.github/workflows/reports-scheduler-test-and-build-workflow.yml

```diff
@@ -3,13 +3,25 @@ name: Test and Build Reports Scheduler
 on: [push, pull_request]
 
 jobs:
+  Get-CI-Image-Tag:
+    uses: opensearch-project/opensearch-build/.github/workflows/get-ci-image-tag.yml@main
+    with:
+      product: opensearch
+
   linux-build:
+    needs: Get-CI-Image-Tag
     strategy:
       matrix:
         java:
           - 11
           - 17
     runs-on: ubuntu-latest
+    container:
+      # using the same image which is used by opensearch-build team to build the OpenSearch Distribution
+      # this image tag is subject to change as more dependencies and updates will arrive over time
+      image: ${{ needs.Get-CI-Image-Tag.outputs.ci-image-version-linux }}
+      # need to switch to root so that github actions can install runner binary on container without permission issues.
+      options: --user root
```

Comment:
where is the install binary logic? root will introduce permission issues for opensearch

Reply:
What binaries are you talking about here?
The image already have all the dependencies needed for running all the build and test.

Reply:
And we are switching to user 1000 when running those gradle commands so there will not be any permission issues.

Reply:
in the comment,
> need to switch to root so that github actions can install runner binary

i saw `su` but it might not be obvious why or when it is needed to switch to normal user

Reply:
linking github doc ref here https://docs.github.com/en/actions/creating-actions/dockerfile-support-for-github-actions#user

> Docker actions must be run by the default Docker user (root). Do not use the USER instruction in your Dockerfile, because you won't be able to access the GITHUB_WORKSPACE directory. For more information, see "[Variables](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables)" and [USER reference](https://docs.docker.com/engine/reference/builder/#user) in the Docker documentation.